Response Logger Spring Boot Starter

TheRagerghost opened the pull request:
Два аспекта были очень сильно завязаны на оригинальном проекте, поэтому перенесены не были.

Для обработчика ошибок была сделана аннотация @LoggableService - раньше было завязано на конкретный сервис в конкретном пакете

Добавил логгинг запросов-ответов. Т.к. нужен веб-контекст, аннотации вешаются на контроллер.

Настройки: 
```yaml
request-logger:
  enabled: true
  logging-level: INFO
```

Для них сделал отдельные подсказки через "additional-spring-configuration-metadata.json"

TheRagerghost committed:
mod: basic starter setup

TheRagerghost committed:
INTERMEDIATE COMMIT

feat: configuration setup
feat: basic logging aspect / annotations

note: previous project had annotation and method to forbid requests, it's not included bc it doesn't fit "logging"

TheRagerghost committed:
INTERMEDIATE COMMIT

feat: @LogExecTime to debug method execution time
feat: @LoggableService to log errors on service methods
feat: @LoggableController to log requests and responses on an annotated controller
feat: @LoggableControllerMethod - same thing, but for annotating a specific method
feat: @LoggableRequest and @LoggableResponse - same thing, but split to log only request on a method or its response

TheRagerghost committed:
INTERMEDIATE COMMIT

debug: slf4j behaviour did not match expectation
fix: switched slf4j for System.out.println() as it's enough for current task
fix: fixed some package paths in files

PROBLEM: look assets/lang-ru-problem-output.png

TheRagerghost committed:
FEAT: RESPONSE LOGGER

fix: switched logs "ru" >> "en"
feat: added ObjectMapper to deserialize request/response body
mod: updated README.md

Alexandr056 commented on src/main/java/com/avragerghost/request_logger_aop/autoconfigure/RequestLoggingAutoConfiguration.java:
```diff
@@ -0,0 +1,16 @@
+import com.avragerghost.request_logger_aop.config.LoggingConfig;
+
+@AutoConfiguration
+@ConditionalOnClass(LoggingAspect.class)
```
думаю это лишнее, конфигурация не зависит от аспекта

TheRagerghost replied:
убрал класс

Alexandr056 commented on src/main/java/com/avragerghost/request_logger_aop/autoconfigure/RequestLoggingAutoConfiguration.java:
```diff
@@ -0,0 +1,16 @@
+@AutoConfiguration
+@ConditionalOnClass(LoggingAspect.class)
+@ConditionalOnProperty(prefix = "request-logger", name = "enabled", havingValue = "true", matchIfMissing = false)
+@Import(LoggingConfig.class)
```
зачем?

TheRagerghost replied:
убрал класс

Alexandr056 commented on src/main/java/com/avragerghost/request_logger_aop/config/LoggingConfig.java:
```diff
@@ -0,0 +1,32 @@
+    }
+
+    @Bean
+    public ObjectMapper objectMapper() {
```
не нужен

TheRagerghost replied:
теперь создаю прямо в аспекте, не бином

Alexandr056 commented on src/main/java/com/avragerghost/request_logger_aop/config/LoggingConfig.java:
```diff
@@ -0,0 +1,32 @@
+@Configuration
+@EnableConfigurationProperties(LoggingProperties.class)
+@EnableAspectJAutoProxy
+public class LoggingConfig {
```
а что мешает его указать для стартера вместо RequestLoggingAutoConfiguration ?

TheRagerghost replied:
сделал

Alexandr056 commented on pom.xml:
```diff
@@ -0,0 +1,59 @@
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-aop</artifactId>
+        </dependency>
+        <dependency>
```
очень много для стартера

TheRagerghost replied:
убрал autoconfigure, т.к. она единственная лишняя оказалась теперь.

starter / configuration-processor / starter-aop нужны в любом случае
starter-web для обработки данных запроса в аспекте

Я смотрел - 15КБ весил стартер, сейчас 13КБ. Думал достаточно, раз даже до МБ не раздувает.

Alexandr056 replied:
много не в том что даже по весу, а что целый веб! томкат, контейнер сервлетов, в общем много много того, что не нужно вообще)

TheRagerghost committed:
FIXES

remove: RequestLoggingAutoConfiguration
remove: autoconfigure dependency

mod: moved ObjectMapper from LoggingConfig (as Bean) to just instantiating it in LoggingAspect
mod: slightly changed message coloring

Alexandr056 commented on src/main/java/com/avragerghost/request_logger_aop/aspects/LoggingAspect.java:
```diff
@@ -0,0 +1,160 @@
+            return;
+        }
+        String methodName = joinPoint.getSignature().getName();
+        ServletRequestAttributes attributes = (ServletRequestAttributes) RequestContextHolder.getRequestAttributes();
```
проще, joinPoint есть у него есть параметры, на него и опирайтесь, там ведь тоже есть тело и заголовки

Alexandr056 commented on src/main/java/com/avragerghost/request_logger_aop/aspects/LoggingAspect.java:
```diff
@@ -0,0 +1,160 @@
+    }
+
+    private boolean shouldLog(LoggingLevel level) {
+        if (!properties.isEnabled()) {
```
не нужен, ибо если флаг выключен бин не создается, поэтому проверка не нужна

Alexandr056 commented on src/main/java/com/avragerghost/request_logger_aop/aspects/LoggingAspect.java:
```diff
@@ -0,0 +1,160 @@
+import jakarta.servlet.http.HttpServletRequest;
+
+@Aspect
+public class LoggingAspect {
```
вместо System.out.println уже логгер)

Alexandr056 commented on src/main/java/com/avragerghost/request_logger_aop/config/LoggingConfig.java:
```diff
@@ -0,0 +1,26 @@
+@Configuration
+@EnableConfigurationProperties(LoggingProperties.class)
+@EnableAspectJAutoProxy
+public class LoggingConfig {
```
не хватает ConditionalOnProperty

Alexandr056 reviewed (changes requested):
хорошая работа над ошибками, но есть над чем еще поработать
